docs/source/java/memory.rst

```diff
@@ -288,6 +288,49 @@ Finally, enabling the ``TRACE`` logging level will automatically provide this st
    |        at RootAllocator.close (RootAllocator.java:29)
    |        at (#8:1)
 
+A further option would be to change the way how allocators are created, we could try the following:
+
+1. Set up a global root allocator that is going to be used in each module,
+2. Create child allocators by using the root allocator,
+3. Give child allocations proper names so that you can detect errors in another class or method,
+4. Ensure that resources are properly closed,
+5. At some strategic point, check if the global allocator is empty,
+6. In case of error, try to review the above-identified class/method allocator bugs.
+
+.. code-block:: java
+
+    //1
+    private static final RootAllocator allocator = new RootAllocator();
```

Comment:
```suggestion
    private static final BufferAllocator allocator = new RootAllocator();
```

You generally should never declare RootAllocator as the type of a variable or parameter.